libflux/flux-core/src/semantic/sub.rs

```diff
@@ -231,6 +264,12 @@ pub trait Substitutable {
     }
 }
 
+impl Substitutable for String {
```

Comment:
It is due to `Property` being tweaked https://github.com/influxdata/flux/pull/4388/files#diff-d5ab296ff16df9447d31cf953c8c275a43922b7cd3b260af30e0325abbdd53eaR1886-R1889 . Since keys may contain variables they need to be `Substitutable` which forces `String` to be `Substitutable`.